lib/flop.ex

```diff
@@ -2004,13 +2027,31 @@ defmodule Flop do
   defp get_order_direction(directions, index),
     do: Enum.at(directions, index, :asc)
 
-  defp new_order_direction(0, :asc), do: :desc
-  defp new_order_direction(0, :asc_nulls_first), do: :desc_nulls_last
-  defp new_order_direction(0, :asc_nulls_last), do: :desc_nulls_first
-  defp new_order_direction(0, :desc), do: :asc
-  defp new_order_direction(0, :desc_nulls_first), do: :asc_nulls_last
-  defp new_order_direction(0, :desc_nulls_last), do: :asc_nulls_first
-  defp new_order_direction(_, _), do: :asc
+  @asc_desc_defaults %{
+    asc: :desc,
+    asc_nulls_first: :desc_nulls_last,
+    asc_nulls_last: :desc_nulls_first,
+    desc: :asc,
+    desc_nulls_first: :asc_nulls_last,
+    desc_nulls_last: :asc_nulls_first
+  }
+
+  defp new_order_direction(0, current_direction, nil),
+    do: @asc_desc_defaults[current_direction]
```

Comment:
I considered naming this `@asc_desc_toggle_map`, as I feel like it's used for the "toggle" case (when the item is the 0-indexed item with no direction overrides). I *feel*, in this moment, that that's a better name and I should have gone with it, but it's a moot point if you don't like the map approach. Let me know!